### PR TITLE
Rename on execution::schedule_from, rename just_on to just_transfer, and add transfer

### PR DESCRIPTION
**Pull request by msimberg**
Continues from #5644.

### CODE REVIEW

**Review by hkaiser — APPROVED** [review]
LGTM, thanks!

**Comment by hkaiser** [comment]
This can go in once the missing replacings have been done.

**Review by aurianer — APPROVED** [review]
LGTM thanks

**Comment by msimberg** [comment]
retest cscs

File: libs/core/execution/include/hpx/execution/algorithms/schedule_from.hpp [hunk 1]
```diff
@@ -292,46 +291,38 @@ namespace hpx { namespace execution { namespace experimental {
...
             {
                 return {s.predecessor_sender, s.scheduler,
                     HPX_FORWARD(Receiver, receiver)};
             }
         };
     }    // namespace detail
 
-    inline constexpr struct on_t final
-      : hpx::functional::detail::tag_fallback<on_t>
+    HPX_HOST_DEVICE_INLINE_CONSTEXPR_VARIABLE struct schedule_from_t final
```
Comment (msimberg) [inline libs/core/execution/include/hpx/execution/algorithms/schedule_from.hpp:310]: `nvcc` was complaining about this not being a device variable, which is why I've changed this to be `HPX_HOST_DEVICE_INLINE_CONSTEXPR_VARIABLE`. However, it's not clear to me what triggers it. Other CPOs are fine as plain `inline constexpr` variables :/ In principle they could all probably be host-device variables, but I'm changing only this one in this PR as a minimal change to make this compile with `nvcc`.

**Comment by msimberg** [comment]
bors merge

**Comment by msimberg** [comment]
bors merge

**Pull request merged**